lint/util.py

```diff
@@ -908,20 +908,30 @@ def find_windows_python(version):
     """Find the nearest version of python and return its path."""
 
     if version:
-        # On Windows, there may be no separately named python/python3 binaries,
-        # so it seems the only reliable way to check for a given version is to
-        # check the root drive for 'Python*' directories, and try to match the
+        from . import persist
+
+        # Try to find python.exe using py.exe
+        # Try the exact version first, then the major version
+        code = r'import sys;print(sys.executable)'
```

Comment:
Instead of using print() and then using splitlines() to cater for added newline, you could just use sys.stdout.write().